### PR TITLE
Automation for COMMERCE-12429 and COMMERCE-12427

### PR DESCRIPTION
**Pull request by Brunno0liveira**
https://liferay.atlassian.net/browse/COMMERCE-12429

https://liferay.atlassian.net/browse/COMMERCE-12427

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 1]
```diff
@@ -14,6 +14,635 @@ definition {
...
+				index = 2,
+				targetFragmentName = "Add to Cart");
+
+			CommerceDisplayPageTemplates.addFragment(
+				collectionName = "Product",
+				fragmentName = "Availability",
+				index = 2,
+				targetFragmentName = "Price");
+
+			CommerceDisplayPageTemplates.addFragment(
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:61]: Add another dynamic field that is set to check the Inventory

Thanks!.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 2]
```diff
@@ -14,6 +14,635 @@ definition {
...
+
+			for (var uomName : list "UOM2,UOM3,UOM4") {
+				AssertElementNotPresent(
+					locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECTOR_VALUE",
+					uomName = ${uomName});
+			}
+		}
+
+		task ("And the UOM data(price, quantity, availability) is shown correctly") {
+			FrontStore.assertProductDetails(
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:164]: Add a check to verify the Inventory.

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 3]
```diff
@@ -14,6 +14,635 @@ definition {
...
+					uomName = ${uomNotPresent});
+			}
+		}
+
+		task ("And assert that the pre-selected UOM’s(UOM2) data is shown correctly") {
+			AssertSelectedLabel(
+				locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECT",
+				value1 = "UOM2");
+
+			FrontStore.assertProductDetails(
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:226]: Add a check to verify the Inventory.

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 4]
```diff
@@ -14,6 +14,635 @@ definition {
...
+			Click(locator1 = "CommerceAccelerators#GENERAL_ADD_TO_CART_BUTTON_ENABLED");
+		}
+
+		task ("When the buyer selects the UOM3") {
+			Select(
+				locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECT",
+				value1 = "UOM3");
+		}
+
+		task ("And the UOM data(price, quantity, availability) is shown correctly") {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:245]: Add a check to verify the Inventory.

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 5]
```diff
@@ -14,6 +14,635 @@ definition {
...
+				key_text = "View all variants",
+				locator1 = "Button#ANY");
+		}
+
+		task ("When the buyer navigates to Minium catalog → MIN55861 product details") {
+			CommerceAcceleratorsInitializer.gotoMiniumProductDetailsPage(productName = "U-Joint");
+		}
+
+		task ("And assert that the information shown on the page is related to the selected SKU/UOM(availability, stock quantity, list price, promo price)") {
+			FrontStore.assertProductDetails(
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:392]: Add checks for stock quantity(add a Dynamic Field showing Inventory) and Promo Price

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 6]
```diff
@@ -14,6 +14,635 @@ definition {
...
+		}
+
+		task ("And selects the second UOM") {
+			Select(
+				locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECT",
+				value1 = "UOM2");
+		}
+
+		task ("And assert that the information shown on the page is related to the selected SKU/UOM(availability, stock quantity, list price, promo price)") {
+			FrontStore.assertProductDetails(
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:412]: Add checks for stock quantity(add a Dynamic Field showing Inventory) and Promo Price

Thanks!

**Review by pietrobocale — CHANGES_REQUESTED** [review]
Review completed take a look at the comments

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 7]
```diff
@@ -14,6 +14,635 @@ definition {
 		CommerceConfiguration.commerceTearDown();
 	}
 
+	@description = "COMMERCE-12429. As admin, I can add to the cart a multi SKUs product with UOM using a Cart Fragment."
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:17]: As buyer

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase [hunk 8]
```diff
@@ -14,6 +14,635 @@ definition {
...
+			}
+		}
+
+		task ("And from the MIN55861 product configuration, multiple and minimum quantity is set as 0.0001") {
+			CommerceJSONProductsAPI._patchCommerceProductConfiguration(
+				minOrderQuantity = "0.0001",
+				multipleOrderQuantity = "0.0001",
+				productName = "Brake Fluid");
+		}
+
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/SFCommerceUnitsOfMeasure2.testcase:138]: Add the buyer login

Thanks!

**Pull request closed without merging**